pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -471,7 +471,9 @@ func (r *hibernationReconciler) checkClusterStopped(cd *hivev1.ClusterDeployment
 		if err := r.updateClusterDeploymentStatus(cd, logger); err != nil {
 			return reconcile.Result{}, err
 		}
-		// logging with time since ready condition was set to StoppingOrHibernating state
+		// Logging with time since ready condition was set to StoppingOrHibernating state
+		// Fetch ready condition again to ensure it points at the correct value
+		readyCondition := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterReadyCondition)
```

Comment:
Note to self: The bug is that L467's `setCDCondition` calls `SetClusterDeploymentConditionWithChangeCheck` which calls `SortClusterDeploymentConditions`. An existing condition pointer points to some index in the array of conditions, so if that array is reordered, the condition at that index can change.